### PR TITLE
Handle cards scheduled for cancellation.

### PR DESCRIPTION
**Pull request by dwybourn**
Added a scheduler which gets the claims which have had a card status of PENDING_CANCELLATION for 16 weeks or more. The scheduler calls a job which sends a notification email and sets the card status to SCHEDULED_FOR_CANCELLATION.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    /**
+     * Notifies claimants that their card is soon to be cancelled and sets the claim's card status to SCHEDULED_FOR_CANCELLATION.
+     * @param claim the claim whose's card is to be scheduled for cancellation
+     */
+    @Transactional
+    public void handleCardPendingCancellation(Claim claim) {
+        MessagePayload messagePayload = buildEmailMessagePayloadWithFirstAndLastNameOnly(claim, CARD_IS_ABOUT_TO_BE_CANCELLED);
+        messageQueueClient.sendMessage(messagePayload, SEND_EMAIL);
+        claim.updateCardStatus(SCHEDULED_FOR_CANCELLATION);
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java]: Maybe not for this PR, but there's no reason why we couldn't send a message to cancel that card at this point too. We could just delay that message by 7 days (or rather, by a configurable period).

File: api/src/main/resources/application-local-development.yml [hunk 2]
```diff
@@ -13,3 +13,11 @@ payment-cycle:
     maximum-lock-time: "PT2M"
 
 enable-scheduler: true
+
+card-cancellation:
+  grace-period: P16W
+  schedule:
+    # run every hour
+    cron-schedule: "*/5 * * * * *"
```
Comment (YetAnotherMatt) [inline api/src/main/resources/application-local-development.yml]: This is every 5 seconds, not hourly. Think it should be: 0 0 * * * *

File: api/src/main/resources/application.yml [hunk 3]
```diff
@@ -85,6 +85,15 @@ payment-cycle:
     maximum-lock-time: "PT59M"
     end-date-offset-days: -1
 
+card-cancellation:
+  grace-period: P16W
+  schedule:
+    # run every hour
+    cron-schedule: "* * */1 * * *"
```
Comment (YetAnotherMatt) [inline api/src/main/resources/application.yml]: shouldn't this be: `0 0 * * * *`?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+@AllArgsConstructor
+@Slf4j
+public class HandleCardPendingCancellationJob {
+
+    private MessageQueueClient messageQueueClient;
+    private final ClaimRepository claimRepository;
+
+    /**
+     * Notifies claimants that their card is soon to be cancelled and sets the claim's card status to SCHEDULED_FOR_CANCELLATION.
+     * @param claim the claim whose's card is to be scheduled for cancellation
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java]: Minor - whose, not whose's

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java (hunk 1 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java]: Definitely sounds like another PR. In fact, shouldn't that be another story seeing as we don't even have an endpoint for cancelling a card yet?

File: api/src/main/resources/application.yml (hunk 3 above)
Comment (dwybourn) [reply api/src/main/resources/application.yml]: Updated to  0 0 * * * *

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java (hunk 4 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java]: Fixed.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJob.java]: That's not covered by this story. I can add a TODO in the next PR linking to HTBHF-1758.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJobTest.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    public void shouldSendCardToBeCancelledEmailAndSetCardStatus() {
+        Claim claim = aClaimWithCardStatusAndCardStatusTimestamp(PENDING_CANCELLATION, LocalDateTime.now().minusWeeks(16));
+
+        job.handleCardPendingCancellation(claim);
+
+        assertEmailSent(claim);
+        assertClaimCardStatusUpdated();
+    }
+
+    private void assertEmailSent(Claim claim) {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJobTest.java]: Is there a nice way we can combine this with the assertions in `DetermineEntitlementNotificationHandlerTest` as they're basically the same apart from the EmailType?

Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJobTest.java]: Refactored common code into EmailPayloadAssertions.

**Pull request merged**